File: data/readable_format/cpp/file364.cpp
#include <iostream>
#include <string>

// Assuming great_rsd simply adds two numbers for demonstration purposes
template<typename T>
T great_rsd(T a, T b) {
    return a + b;
}

// A simple assert function for testing
template<typename T>
void assert_equals(T expected, T actual, const std::string& test_name) {
    if (expected == actual) {
        std::cout << test_name << " passed." << std::endl;
    } else {
        std::cout << test_name << " failed. Expected: " << expected << " Got: " << actual << std::endl;
    }
}

int main() {
    // Test cases for great_rsd
    assert_equals(8, great_rsd(5, 3), "Test 1: Adding positive numbers");
    assert_equals(0, great_rsd(-1, 1), "Test 2: Adding negative and positive numbers");
    assert_equals(-15, great_rsd(-5, -10), "Test 3: Adding negative numbers");

    return 0;
}

File: data/readable_format/fortran/file364.f90
program test_great_rsd
    implicit none

    ! Test variables
    integer :: a, b, result

    ! Test cases
    a = 5; b = 3
    result = great_rsd(a, b)
    call assert_equals(8, result, "Test 1")

    a = -1; b = 1
    result = great_rsd(a, b)
    call assert_equals(0, result, "Test 2")

    a = -5; b = -10
    result = great_rsd(a, b)
    call assert_equals(-15, result, "Test 3")

contains

    function great_rsd(a, b)
        integer, intent(in) :: a, b
        integer :: great_rsd
        great_rsd = a + b
    end function great_rsd

    subroutine assert_equals(expected, actual, message)
        integer, intent(in) :: expected, actual
        character(len=*), intent(in) :: message
        if (expected /= actual) then
            print *, trim(message) // " failed. Expected: ", expected, " Got: ", actual
        else
            print *, trim(message) // " passed."
        endif
    end subroutine assert_equals

end program test_great_rsd